### PR TITLE
[silgenpattern] NFC cleanups in preparation for fixing fallthrough issue

### PR DESCRIPTION
**Pull request by gottesmm**
This just consists of 3 NFC commits that fix up the code a bit. I am doing this so that #23056 is easier to review by upstreaming these first.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

File: lib/SILGen/SILGenPattern.cpp [hunk 1]
```diff
@@ -2437,64 +2439,69 @@ void PatternMatchEmission::emitSharedCaseBlocks() {
...
+    // Make sure that before/after we emit the body we have the same cleanups
+    // depth to ensure that all cleanups from inside the case body have been
+    // popped.
     assert(SGF.getCleanupsDepth() == PatternMatchStmtDepth);
+    SWIFT_DEFER { assert(SGF.getCleanupsDepth() == PatternMatchStmtDepth); };
+
+    // If we do not have any bound decls, just emit the case body... there is no
+    // further work to do.
+    if (!caseBlock->hasBoundDecls()) {
+      return emitCaseBody(caseBlock);
```
Comment (gottesmm) [inline lib/SILGen/SILGenPattern.cpp]: This was why this originally failed. I should have used a continue instead of a return.

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Review by shahzadlone — APPROVED** [review]

**Pull request merged**